crates/iota-indexer/src/store/token_unlocks.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{fs, path::PathBuf};
+
+use chrono::{DateTime, Utc};
+use csv::ReaderBuilder;
+
+use crate::models::token_unlocks::{Entity, TokenUnlock};
+
+/// In-memory store holding all token unlock schedules across entities.
+/// Allows querying how many tokens are currently locked.
+#[derive(Debug, Clone)]
+pub struct TokenUnlocksStore {
+    /// All known entities with their respective unlock schedules.
+    entities: Vec<Entity>,
+}
+
+impl TokenUnlocksStore {
+    /// Loads all token unlock data from CSV files located in the
+    /// `token_unlocks_data` directory at the crate root. Each CSV file
+    /// represents the unlock schedule for a single entity.
+    ///
+    /// Panics if the directory is missing or unreadable.
+    pub fn load() -> Self {
+        let crate_dir = env!("CARGO_MANIFEST_DIR");
+        let dir = PathBuf::from(crate_dir).join("token_unlocks_data");
+
+        let paths = fs::read_dir(&dir)
+            .unwrap_or_else(|_| panic!("Token Unlock CSV directory not found: {:?}", dir));
+
+        let entities = paths
+            .filter_map(|entry| entry.ok())
+            .filter(|e| e.path().extension().unwrap_or_default() == "csv")
+            .map(|entry| {
+                let path = entry.path();
+                let file_name = path.file_stem().unwrap().to_string_lossy().to_string();
+                let file = fs::File::open(&path)
+                    .unwrap_or_else(|_| panic!("Failed to open file {:?}", path));
```

Comment:
You can do some cleanup like this in more places
```suggestion
            .unwrap_or_else(|_| panic!("Token Unlock CSV directory not found: {dir:?}"));

        let entities = paths
            .filter_map(|entry| entry.ok())
            .filter(|e| e.path().extension().unwrap_or_default() == "csv")
            .map(|entry| {
                let path = entry.path();
                let file_name = path.file_stem().unwrap().to_string_lossy().to_string();
                let file = fs::File::open(&path)
                    .unwrap_or_else(|_| panic!("Failed to open file {path:?}"));
```

Reply:
This code has been changed with the latest commit, should be fine now